src/main/scala/tech/sourced/gemini/Gemini.scala

```diff
@@ -58,30 +61,52 @@ object Gemini {
       //TODO: implement based on Apolo
       //findSimilarProjects(path, conn)
     } else {
-      findDuplicateFiles(path, conn)
+      findDuplicateBlobHash(path, conn)
       //TODO: implement based on Apolo
       //findSimilarFiles(path, conn)
     }
   }
 
   /**
-    * Finds duplicated files among hashed repositories that already exists in certain repository
+    * Finds duplicate files among hashed repositories
+    *
+    * @param conn   Database connection
+    * @return
+    */
+  def report(conn: Session): Iterable[DuplicateBlobHash] = {
+    findDuplicateBlobHash(conn)
+  }
+
+  /**
+    * Finds duplicate files among hashed repositories, and returns all duplicated files
     *
-    * @param repository repository url, example: github.com/src-d/go-git.git"
+    * @param conn   Database connection
     * @return
     */
-  def report(repository: String): Iterable[RepoFile] = {
-    throw new UnsupportedOperationException("Finding all duplicate files in many repositories is no implemented yet.")
+  def detailedReport(conn: Session): Iterable[Iterable[RepoFile]] = {
+    findDuplicateBlobHash(conn)
+        .map {d => findDuplicateBlobHash(d.sha, conn)}
+  }
+
+  def findDuplicateBlobHash(conn: Session): Iterable[DuplicateBlobHash] = {
+    val cql = "SELECT blob_hash, repo, file_path, COUNT(*) as count FROM hashes.blob_hash_files GROUP BY blob_hash"
+    conn
+      .execute(new SimpleStatement(cql))
+      .asScala
+      .filter( _.getLong("count") > 1 )
+      .map { r => DuplicateBlobHash(r.getString("blob_hash"), r.getLong("count")) }
   }
 
   def findDuplicateProjects(in: File, conn: Session): Iterable[RepoFile] = {
     //TODO(bzz): project is duplicate if it has all it's files in some other projects
     throw new UnsupportedOperationException("Finding duplicate repositories is no implemented yet.")
   }
 
-  def findDuplicateFiles(file: File, conn: Session): Iterable[RepoFile] = {
-    val sha = computeSha1(file)
+  def findDuplicateBlobHash(file: File, conn: Session): Iterable[RepoFile] = {
+    findDuplicateBlobHash(computeSha1(file), conn)
+  }
 
+  def findDuplicateBlobHash(sha: String, conn: Session): Iterable[RepoFile] = {
```

Comment:
Having two overriden methods with the same names that have only different argument types usually is not a good idea - it does not help a reader to understand the purpose or difference of those methods.

One suggesion from above is to have `findAllDuplicateItems` for listing the dups and then `findDuplicateHashesFor` a given item/blobHash